features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -326,7 +343,10 @@ void ATHandler::process_oob()
         _at_timeout = timeout;
         tr_debug("AT OoB done");
     }
-    unlock();
+    _event_id = -1;
```

Comment:
@kjbracey-arm, this was why _event_id could be negative. This should use 0 as an invalid value.